src/lib.rs

```diff
@@ -0,0 +1,111 @@
+extern crate tokio_trace;
+
+struct Span {
+    label: String,
+    ref_count: usize,
+}
+
+use std::fmt;
+use std::sync::Mutex;
+
+use tokio_trace::field::{Field, Visit};
+use tokio_trace::metadata::Level;
+use tokio_trace::{span, Event, Metadata, Subscriber};
+pub struct Visitor;
+
+impl Visit for Visitor {
+    fn record_debug(&mut self, field: &Field, value: &fmt::Debug) {
+        if field.name() == "message" {
+            print!("{:?}: ", value);
+        } else {
+            print!("{} = {:?} ", field.name(), value);
+        }
+    }
+}
+
+pub struct DemoSubscriber {
+    spans: Mutex<Vec<Span>>,
+    stack: Mutex<Vec<u64>>,
+}
+
+impl DemoSubscriber {
+    pub fn new() -> DemoSubscriber {
+        DemoSubscriber {
+            spans: Mutex::default(),
+            stack: Mutex::default(),
+        }
+    }
+}
+
+impl Subscriber for DemoSubscriber {
+    fn enabled(&self, _metadata: &Metadata) -> bool {
+        true
+    }
+
+    fn new_span(&self, attributes: &span::Attributes) -> span::Id {
+        // TODO: correlate spans based on their metadata
+        let mut spans = self.spans.lock().unwrap();
+        let id = spans.len() + 1;
+        spans.push(Span {
+            label: attributes.metadata().name().to_string(),
```

Comment:
Note that in addition to metadata, a span's `Attributes` will also contain any fields that were provided when the span was created. You can record these fields here by passing a visitor to the [`Attributes::record`](https://docs.rs/tokio-trace-core/0.1.0/tokio_trace_core/span/struct.Attributes.html#method.record) function.